src/client/datascience/jupyter/kernels/cellExecutionQueue.ts

```diff
@@ -33,22 +40,31 @@ export class CellExecutionQueue {
     }
     constructor(
         private readonly notebookPromise: Promise<INotebook>,
-        private readonly executionFactory: CellExecutionFactory,
+        private readonly cellExecutionFactory: CellExecutionFactory,
+        private readonly codeExecutionFactory: CodeExecutionFactory,
         readonly metadata: Readonly<KernelConnectionMetadata>
     ) {}
     /**
      * Queue the cell for execution & start processing it immediately.
      */
-    public queueCell(cell: NotebookCell): void {
-        const existingCellExecution = this.queueOfCellsToExecute.find((item) => item.cell === cell);
-        if (existingCellExecution) {
-            traceCellMessage(cell, 'Use existing cell execution');
-            return;
-        }
-        const cellExecution = this.executionFactory.create(cell, this.metadata);
-        this.queueOfCellsToExecute.push(cellExecution);
+    public queueCell(cell?: NotebookCell, code?: string): void {
+        if (cell) {
+            const cellQueue = this.getCellExecutions(this.queueToExecute);
+            const existingCellExecution = cellQueue.find((item) => item.cell === cell);
+            if (existingCellExecution) {
+                traceCellMessage(cell, 'Use existing cell execution');
+                return;
+            }
+            const cellExecution = this.cellExecutionFactory.create(cell, this.metadata);
+            this.queueToExecute.push({ cellExecution });
 
-        traceCellMessage(cell, 'User queued cell for execution');
+            traceCellMessage(cell, 'User queued cell for execution');
+        } else if (code) {
+            const codeExecution = this.codeExecutionFactory.create(code);
+            this.queueToExecute.push({ codeExecution });
+
+            traceInfo('Hidden cell queued for execution', codeExecution.code.substring(0, 50));
+        }
 
```

Comment:
We have missed an else clause and that results in bugs, the suggestion I made will ensure typescript types will prevent such bugs